extensions/src/test/java/org/pentaho/test/platform/web/http/security/RequestParameterAuthenticationFilterTest.java

```diff
@@ -97,4 +109,125 @@ public void userNamePasswordUnencrypted() throws IOException, ServletException {
     verify( authManagerMock ).authenticate( Mockito.eq( authRequest ) );
 
   }
+
+
+  @Test
+  public void testSetCsrfProcessorRespectsIt() {
+
+    CsrfProcessor csrfProcessorMock = mock( CsrfProcessor.class );
+    RequestParameterAuthenticationFilter filter = new RequestParameterAuthenticationFilter();
+
+    // ---
+
+    filter.setCsrfProcessor( csrfProcessorMock );
+
+    // ---
+
+    assertSame( csrfProcessorMock, filter.getCsrfProcessor() );
+  }
+
+  private void testRequestParameterAuthenticationFailsThenThrowsCsrfValidationAuthException( @NonNull Throwable validateError )
+          throws ServletException, IOException {
+
+    CsrfProcessor csrfProcessorMock = mock( CsrfProcessor.class );
+
+    final MockHttpServletRequest request =
+            new MockHttpServletRequest( "GET",
+                    "http://localhost:9080/pentaho-di/kettle/executeTrans/?rep=dev&userid=admin&password=password&trans=/home"
+                            + "/admin/Trans1" );
+    request.addParameter( "userid", "admin" );
+    request.addParameter( "password", "password" );
+
+    filter.setAuthenticationManager( authManagerMock );
+    filter.setCsrfProcessor( csrfProcessorMock );
+
+    when( csrfProcessorMock.validateRequestOfVulnerableOperation( eq( request ), anyString() ) )
+            .thenThrow( validateError );
+
+    // ---
+
+    try {
+      filter.doFilter( request, new MockHttpServletResponse(), new MockFilterChain() );
+      fail( "Should have thrown exception" );
+    } catch ( CsrfValidationAuthenticationException ex ) {
```

Comment:
The `CsrfValidationAuthenticationException` exception was created for use in the `CsrfProtectedUsernamePasswordAuthenticationFilter`, but is not being used here (and need not be used here).

Besides, when the CSRF processor throws, one exception is being caught and handled by calling the `authenticationEntryPoint` and the other two exceptions are not being caught.

To test the caught exception, you'd check that the `authenticationEntryPoint` was called.
To test the not caught exceptions, the test case would be marked as, e.g.: `@Test(expected=IOException.class)`